src/generic/stage1/utf8_lookup4_algorithm.h

```diff
@@ -162,6 +162,12 @@ using namespace simd;
       this->error |= this->prev_incomplete;
     }
 
+#if __cplusplus > 201703L || (defined(_MSVC_LANG) && _MSVC_LANG > 201703L)
+#define IF_CONSTEXPR if constexpr
```

Comment:
Please prefix it:

`SIMDJSON_IF_CONSTEXPR`.

It might also be interesting to guard against already defined macros.